python/pyarrow/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -296,7 +296,7 @@ class NumPyConverter {
   PyArrayObject* mask_;
   int64_t length_;
   int64_t stride_;
-  int itemsize_;
+  int64_t itemsize_;
```

Comment:
@seberg question here, I took this from your suggestion, but from reading the numpy PR, this field is changing from `int` to `intp`, not `int64`?

Reply:
(our windows build is complaining about this, `warning C4244: 'argument': conversion from 'int64_t' to 'int32_t', possible loss of data`)

Reply:
Hmmm, maybe because this line has to be adjusted as well?  https://github.com/apache/arrow/blob/27f15abeba7ffc350896439038ff0b9749111610/python/pyarrow/src/arrow/python/numpy_to_arrow.cc#L566

Unless that is a 32bit system I don't think `intp` vs. `int64` is relevant. I used `int64` because you are using it for shape/strides which are also `intp` in NumPy, but of course `ssize_t` works also (if you have it defined).

Reply:
OTOH, I don't know if that is OK, maybe it should just reject hypothetical very large dtypes?

Reply:
Just to note, since I am not sure where to look :(.  If this is windows specific, it would sound like a `long` is involved (which is 32bit on windows).  But I don't really see where/how.

Reply:
Indeed, another `builder.Append` instance needed a static cast. 

I am not sure if we actually check somewhere the value of itemsize in case that would be larger than the int32 max